src/pivot-table/components/cells/styles.tsx

```diff
@@ -1,21 +1,30 @@
 import Box from "@mui/material/Box";
 import { styled } from "@mui/material/styles";
 import type { stardust } from "@nebula.js/stardust";
+import { HEADER_ICON_SIZE } from "../../constants";
+import { CELL_PADDING, textStyle } from "../shared-styles";
 
-interface Props {
+const baseFlex: React.CSSProperties = {
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+};
+
+interface StyledHeaderCellWrapperProps {
   interactions: stardust.Interactions;
   background: string;
   hoverBackground: string;
+  shouldShowMenuIcon: boolean;
 }
 
 export const StyledHeaderCellWrapper = styled(Box, {
-  shouldForwardProp: (prop: string) => prop !== "interactions" && prop !== "hoverBackground" && prop !== "background",
-})(({ interactions, background, hoverBackground }: Props) => ({
-  padding: 0,
+  shouldForwardProp: (prop: string) =>
+    !["interactions", "hoverBackground", "background", "shouldShowMenuIcon"].includes(prop),
+})(({ interactions, background, hoverBackground, shouldShowMenuIcon }: StyledHeaderCellWrapperProps) => ({
   position: "relative",
   display: "grid",
-  gridTemplateColumns: "1fr 24px",
-  gridGap: "4px",
+  gridTemplateColumns: shouldShowMenuIcon ? `1fr ${HEADER_ICON_SIZE}px` : "1fr",
+  gridGap: shouldShowMenuIcon ? CELL_PADDING : 0,
```

Comment:
Don't think you need the ternary operator here. `gridGap: CELL_PADDING` should work just fine.